src/components/Table/Internal/OcTable.tsx

```diff
@@ -506,7 +509,11 @@ function OcTable<RecordType extends DefaultRecordType>(
       onRow={onRow}
       emptyNode={emptyNode}
       childrenColumnName={mergedChildrenColumnName}
-      onRowHoverEnter={onRowHoverEnter}
+      onRowHoverEnter={(index, key, event) => {
+        const hoveredCell = event.target as HTMLElement;
+        setHoveredRowBoundingRect(hoveredCell.getBoundingClientRect());
```

Comment:
To ensure `<Scroller />` renders when needed create the hover equivalent of `onTriggerExpand` and call it here. That way the row keys are good to go.